server/src/graphql/resolvers/pathway.js

```diff
@@ -50,6 +50,47 @@ const resolver = {
                 return { status: 'ERROR', message: e.toString() }
             }
         },
+        async updateUserStepStatus(_, {username, stepId, status}) {
+            const query = queries.user.updateStepStatus({
+                username, stepId, status
+            })
+
+            try {
+                await query.run()
+                return { status: 'OK', message: null}
+            } catch (e) {
+                return {status: 'ERROR', message: e.toString() }
+            }
+        }
+    },
+    Step: {
+        async userStatus(parent, _, context) {
+            const user = context.user
+            if (user == null) {
+                return "NOT_STARTED"
+            }
+
+            const username = user.username
+            const stepId = parent.id
+            const query = queries.user.getStepStatus({
```

Comment:
You're gonna run a query for every single step? That's not good. Either use a dataloader, or see if you can do something with `delegateToSchema` and GraphQL context. Definitely not this.

Reply:
Yes. Will use a dataloader.